Assets/MixedRealityToolkit/UX/Scripts/Progress/ProgressIndicatorOrbsRotator.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+
+namespace MixedRealityToolkit.UX.Progress
+{
+    public class ProgressIndicatorOrbsRotator : MonoBehaviour
+    {
+        public GameObject[] Orbs;
+
+        public float RotationSpeedRawDegrees;
+
+        public float SpacingDegrees;
+
+        public float Acceleration;
+
+        public int Revolutions;
+
+        public bool TestStop = false;
+
+        public bool HasAnimationFinished = false;
+
+        private float timeElapsed;
+        private int deployedCount;
+        private bool timeUpdated;
+        private float[] angles;
+        private float timeSlice;
+        private float deg2rad = Mathf.PI / 180.0f;
+        private GameObject[] dots;
+        private bool stopRequested;
+        private float rotationWhenStopped;
+
+        // Use this for initialization
+        void Start()
+        {
+            rotationWhenStopped = 0.0f;
+            stopRequested = false;
+            timeSlice = 0;
+            timeUpdated = false;
+            deployedCount = 1;
+            timeElapsed = 0.0f;
+            angles = new float[Orbs.Length];
+            for (int i = 0; i < angles.Length; ++i)
+            {
+                angles[i] = 0;
+            }
+
+            dots = new GameObject[5];
+            for (int i = 0; i < Orbs.Length; ++i)
+            {
+                dots[i] = Orbs[i].transform.GetChild(0).gameObject;
+                dots[i].GetComponent<Renderer>().material.color = new Color(1, 1, 1, 1);
```

Comment:
memory leak when setting color because we don't clean up the created material references.

Reply:
use sharedMaterial